devicetypes/smartenit/iot8-z.src/iot8-z.groovy

```diff
@@ -0,0 +1,223 @@
+					}
+				} else if (eventDescMap?.clusterId == "000C" || eventDescMap?.cluster == "000C") {
+					long convertedValue = Long.parseLong(eventDescMap?.value, 16)
+					Float percentage = Float.intBitsToFloat(convertedValue.intValue())
+					percentage = (percentage / 1.60) * 100.0
+					def ceilingVal = Math.ceil(percentage)
+					if (ceilingVal > 100.0) {
+						ceilingVal = 100.0
+					}
+
+					int intValue = (int) ceilingVal
+
+					def map = createEvent(name: "inputValue", value: intValue)
+					childDevice.sendEvent(map)
+				}
+			}
+		}
+	}
+}
+
+def on() {
+	zigbee.on()
+}
+
+def off() {
+	zigbee.off()
+}
+
+def childOn(String dni) {
+	def childEndpoint = getChildEndpoint(dni)
+	zigbee.command(zigbee.ONOFF_CLUSTER, 0x01, "", [destEndpoint: childEndpoint])
+}
+
+def childOff(String dni) {
+	def childEndpoint = getChildEndpoint(dni)
+	zigbee.command(zigbee.ONOFF_CLUSTER, 0x00, "", [destEndpoint: childEndpoint])
+}
+
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	def refreshCommands = zigbee.onOffRefresh() 
+	def numberOfChildDevices = 8
+
+	for(def endpoint : 2..numberOfChildDevices) {
+		refreshCommands += zigbee.readAttribute(zigbee.ONOFF_CLUSTER, getONOFF_ATTRIBUTE(), [destEndpoint: endpoint])
+	}
+	for(def endpoint : 1..4) {
+		refreshCommands += zigbee.readAttribute(getBINARY_INPUT_CLUETER(), getPRESENT_VALUE_ATTRIBUTE(), [destEndpoint: endpoint])
+	}
+
+	refreshCommands += zigbee.readAttribute(getANALOG_INPUT_CLUSTER(), getPRESENT_VALUE_ATTRIBUTE(), [destEndpoint: 0x0001]);
+	refreshCommands += zigbee.readAttribute(getANALOG_INPUT_CLUSTER(), getPRESENT_VALUE_ATTRIBUTE(), [destEndpoint: 0x0002]);
+	log.debug "refreshCommands: $refreshCommands"
+
+	return refreshCommands
+}
+
+private void createChildDevices() {
+	def numberOfChildDevices = 8
+
+	for (def endpoint: 2..numberOfChildDevices) {
+		try {
+			if (endpoint == 2) {
+				addChildDevice("Smartenit", "IOT8-Z-child-analog-contact-switch", "${device.deviceNetworkId}:0${endpoint}", device.hubId,
+					[completedSetup: true,
+					label: "${device.displayName} ${endpoint}",
+					isComponent: false
+					])
+			} else if (endpoint >= 3 && endpoint <= 4) {
+				addChildDevice("Smartenit", "IOT8-Z-child-contact-switch", "${device.deviceNetworkId}:0${endpoint}", device.hubId,
+					[completedSetup: true,
+					label: "${device.displayName} ${endpoint}",
+					isComponent: false
+					])
+			} else if (endpoint >= 5 && endpoint <= 8) {
+				addChildDevice("smartthings", "ZigBee Switch", "${device.deviceNetworkId}:0${endpoint}", device.hubId,
+					[completedSetup: true,
+					label: "${device.displayName[0..-7]}${endpoint}",
+					isComponent: false
+					])
+			}
+		} catch (Exception e) {
+			log.debug "Exception creating child device: ${e}"
+		}
+	}
+}
+
+def configure() {
+	log.debug "configure"
+    
+	configureHealthCheck()
+	def configurationCommands = zigbee.configureReporting(getBINARY_INPUT_CLUETER(), getPRESENT_VALUE_ATTRIBUTE(), 0x10, 10, 600, null)
```

Comment:
Side note, if you define get method like you did
```
private getANALOG_INPUT_CLUSTER() { 0x000C }
```
you can access value by simple `ANALOG_INPUT_CLUSTER`.